interview-questions.md

```diff
@@ -8,42 +8,88 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: Ruby is a language built around OOP, which is to say that everything in Ruby is an object with unique properties. Treating everything as an object helps streamline code and make it more modular and diverse to work with. Variables and data in OOP can be changed and mutated by the methods that are applied them. Functional programming is more strict in the sense that functions typically produce the same output given the same input. Variables in functional programming are typically declared and not changed as they are used as arguements for functions.
 
-Researched answer:
+Researched answer: OOP is a model of programming where everything is an object that holds data, and within those objects are methods that are used to change the program's state (or memory at a given point in the program's execution). Code in OOP is considered Imperitive, meaning that it describes how a method or program is supposed to work step-by-step. This is useful when adding new data or classes seperate from existing data or classes, we can use methods on the new data without changing the existing data. We define how we want our code to operate, which makes it easier to add new things later on. Functional programming is a form of programming that is centered around building functions that typically do not change state and avoids changing data. An ouput of a function in FP should be the same given the same input, ehich makes FP useful for testing. Variables in FP are considered immutable, which means that ones the variable is declared, it cannot be changed. The only way to change the value of a given variable is to assign the new value to a new variable. In contrast to OOP having an Imperitive model, FP uses a Declarative programming model that describes what the program needs to accomplish, not HOW it should be accomplished. FP allows for parallel computing which can be used to divide large processes into smaller ones that can be solved in tandem with each other, decreases processing time.  
+(sources - https://www.educative.io/blog/functional-programming-vs-oop, https://www.geeksforgeeks.org/difference-between-functional-programming-and-object-oriented-programming/, https://en.wikipedia.org/wiki/Imperative_programming, https://medium.com/@shaistha24/functional-programming-vs-object-oriented-programming-oop-which-is-better-82172e53a526)
 
-1. What is the difference between a Float and an Integer in Ruby?
+2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: A Float is a Class in ruby that is a number with a decimal (1.5, 3.14, 2.0), and an Integer is a Class in ruby that are whole numbers with no decimals (1, 4, 62)
 
-Researched answer:
+**_(outside of doing research on this one, I'm not sure there is much more to say about either with my answer above, apologies if this doesn't suffice)_**
```

Comment:
This is great! Some questions can be answered shortly and don't need drawn out explanations. One of the things we want to consider when working with these different data types, is what happens? For example, if we divide an integer by a float, what will our output be?